resources/js/web3_init.js

```diff
@@ -66,57 +70,19 @@ StatusHttpProvider.prototype.sendAsync = function (payload, callback) {
     else {
         var messageId = callbackId++;
         callbacks[messageId] = callback;
-        if (typeof StatusBridge == "undefined") {
-            var data = {
-                payload: JSON.stringify(payload),
-                callbackId: JSON.stringify(messageId),
-                host: this.host
-            };
-
-            webkit.messageHandlers.sendRequest.postMessage(JSON.stringify(data));
-        } else {
-            StatusBridge.sendRequest(this.host, JSON.stringify(messageId), JSON.stringify(payload));
-        }
-    }
-};
-
-StatusHttpProvider.prototype.prepareRequest = function () {
-    var request = new XMLHttpRequest();
-
-    request.open('POST', this.host, false);
-    request.setRequestHeader('Content-Type', 'application/json');
-    return request;
-};
-
-/**
- * Synchronously tries to make Http request
- *
- * @method isConnected
- * @return {Boolean} returns true if request haven't failed. Otherwise false
- */
-StatusHttpProvider.prototype.isConnected = function () {
-    try {
-        this.sendAsync({
-            id: 9999999999,
-            jsonrpc: '2.0',
-            method: 'net_listening',
-            params: []
-        }, function () {
-        });
-        return true;
-    } catch (e) {
-        return false;
+        WebViewBridge.send(JSON.stringify({type:      'web3-send-async',
+                                           messageId: messageId,
+                                           payload:   payload}));
     }
 };
 }
 
 var protocol = window.location.protocol
-var address = providerAddress || "http://localhost:8545";
-console.log(protocol);
 if (typeof web3 === "undefined") {
+    //why do we need this condition?
```

Comment:
upgrade react-native-webview-bridge (fix preloading of js on the first page, Android)
ensure that preloaded js in webview is executed only once
https://github.com/status-im/status-react/commit/99931c24b924aff2d339a46c7a7ae321f288daac#diff-5db6d4dab305f66fa6f1813ce37dd68d

inject web3 before page loading in webview (works only on android)
https://github.com/status-im/status-react/commit/89eb890d9436d2ff66bef6f8e13e195f95bc5aff#diff-5db6d4dab305f66fa6f1813ce37dd68d

Reply:
sorry @yenda  i mean this one `if (protocol == "https:" || protocol == "http:") {`